pulsar-io/kafka-connect-adaptor/src/main/java/org/apache/pulsar/io/kafka/connect/KafkaConnectSink.java

```diff
@@ -290,4 +299,25 @@ protected long currentOffset(String topic, int partition) {
         return taskContext.currentOffset(topic, partition);
     }
 
+    // Replace all non-letter, non-digit characters with underscore.
+    // Append underscore in front of name if it does not begin with alphabet or underscore.
+    protected String sanitizeNameIfNeeded(String name, boolean sanitize) {
+        if (!sanitize) {
+            return name;
+        }
+
+        try {
+            return sanitizedTopicCache.get(name, () -> {
```

Comment:
I believe that we drop persistent:// as it is only boilerplate (also non-persistent)